src/suction_grasp_data.cpp

```diff
@@ -0,0 +1,109 @@
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/* Authors: Dave Coleman, Bence Magyar
+   Description: Data class used by the grasp generator.
+*/
+
+#include <moveit_grasps/suction_grasp_data.h>
+
+// Eigen
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+#include <eigen_conversions/eigen_msg.h>
+
+// C++
+#include <math.h>
+
+// Parameter loading
+#include <rosparam_shortcuts/rosparam_shortcuts.h>
+
+// Pose conversion
+#include <rviz_visual_tools/rviz_visual_tools.h>
+
+namespace moveit_grasps
+{
+const std::string LOGNAME = "grasp_data.suction_gripper";
+
+SuctionGraspData::SuctionGraspData(const ros::NodeHandle& nh, const std::string& end_effector,
+                                   moveit::core::RobotModelConstPtr robot_model)
+  : GraspData(nh, end_effector, robot_model)
+{
+}
+
+bool SuctionGraspData::loadGraspData(const ros::NodeHandle& nh, const std::string& end_effector)
+{
+  if (!GraspData::loadGraspData(nh, end_effector))
+  {
+    ROS_ERROR_STREAM_NAMED(LOGNAME, "GraspData::loadGraspData failed");
+    return false;
+  }
+  // Overwrite postures since they are not used with suction grippers
+  pre_grasp_posture_.joint_names = {};
+  pre_grasp_posture_.points.resize(1);
+  pre_grasp_posture_.points[0].positions = {};
+
+  grasp_posture_.joint_names = {};
+  grasp_posture_.points.resize(1);
+  grasp_posture_.points[0].positions = {};
+
+  // Load all other parameters
+  const std::string parent_name = "grasp_data";  // for namespacing logging messages
+  std::size_t error = 0;
+
+  // Search within the sub-namespace of this end effector name
+  ros::NodeHandle child_nh(nh, end_effector);
+
+  int suction_rows_count, suction_cols_count;
+  error += !rosparam_shortcuts::get(parent_name, child_nh, "active_suction_range_x", active_suction_range_x_);
+  error += !rosparam_shortcuts::get(parent_name, child_nh, "active_suction_range_y", active_suction_range_y_);
+  child_nh.param<int>("suction_rows_count", suction_rows_count, 1);
+  child_nh.param<int>("suction_cols_count", suction_cols_count, 1);
+  suction_voxel_matrix_ = std::make_shared<SuctionVoxelMatrix>(suction_rows_count, suction_cols_count,
+                                                               active_suction_range_y_, active_suction_range_x_);
+  rosparam_shortcuts::shutdownIfError(parent_name, error);
+
+  return true;
+}
+
+void SuctionGraspData::print()
+{
+  GraspData::print();
+  std::cout << "Suction Gripper Parameters: " << std::endl;
+  std::cout << "\tactive_suction_range_x_: " << active_suction_range_x_ << std::endl;
+  std::cout << "\tactive_suction_range_y_: " << active_suction_range_y_ << std::endl;
```

Comment:
Not sure why the original GraspData class used std::cout here. This was separated out kept consistent with GraspData. This could totally be fixed in a follow up PR